### PR TITLE
Controller spec

### PR DESCRIPTION
**Pull request by Zuwairanajma**
**In this milestone, I implemented the following**
- Added RSpec as a gem to the project.
- Created a Request spec file for all controllers.
- Ensured that for each action the following were checked:
- If the response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- CONTRIBUTING -->
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/Zuwairanajma/Blog-App/issues).
+
```
Comment (MahmoudBakr23) [inline README.md]: - [ ] Please make sure to fix the issues page for a professional README file. 🙏

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    it 'renders the show template' do
+      get '/users/732/posts/2'
+      expect(response).to render_template(:show)
+    end
+
+    it 'contains the post ID in the URL' do
+      get '/users/732/posts/2'
+      expect(request.fullpath).to include('/users/732/posts/2')
+    end
```
Comment (MahmoudBakr23) [inline spec/requests/posts_spec.rb:25]: - [ ] Please consider writing the GET/show testing method in a different block using a new `describe` method to separate the tests and make it easier to identify. 🙏

_Kindly do the same for the GET/show testing method in the users_spec as well._

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @Zuwairanajma 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done writing the controllers' tests so far
- Descriptive PR
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /users/732/posts' do
```
Comment (TedLivist) [inline spec/requests/posts_spec.rb:4]: - [ ] **[OPTIONAL]** I would advise you to use a more descriptive text here to describe what you are testing.

  ```
  describe GET #index
  ```

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      expect(response).to have_http_status(200)
+    end
+
+    it 'renders the index template' do
+      get '/users/732/posts'
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe 'GET /users/732/posts/:id' do
```
Comment (TedLivist) [inline spec/requests/posts_spec.rb:16]: - [ ] **[OPTIONAL]** I would advise you to use a more descriptive text here to describe what you are testing.

  ```
  describe GET #show
  ```

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Zuwairanajma,

✔️ Well-written and passing tests

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**